purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1301,31 +1386,51 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
         }
     }
 
+    @Synchronized
+    private fun getAndClearProductChangeCallback(): ProductChangeListener? {
```

Comment:
yeah, it would work, but be unnecessary (at least in Java, I expect that the same would be true for kotlin https://stackoverflow.com/a/249907/5074358)

Reply:
Yeah, I tend to over use them because they are reentrant, but I will remove this one and the one in `getPurchaseCallback`